.vale.ini

```
@@ -0,0 +1,27 @@
+StylesPath = .github/styles
+
+# https://github.com/errata-ai/packages
+Packages = proselint,\
+           write-good
+
+[*.md]
+BasedOnStyles = proselint,\
+                Vale,\
+                write-good
+
+# Disable
+Vale.Spelling = NO
+write-good.E-Prime = NO
+write-good.Passive = NO
+write-good.TooWordy = NO
+
+# Ignore lines starting with "> [!NOTE]"
+BlockIgnores = (?s)> \[!NOTE\].*?(\n|$)
```

This replaces the inline comment before @matt-graham

---

Eeeeeh what about `TIP` `IMPORTANT` `WARNING` `CAUTION`? Do they get flagged.
I could imagine `> [!TIP]` is quite useful.

```suggestion
BlockIgnores = (?s)> \[!(NOTE|TIP|WARNING|CAUTION|IMPORTANT)\].*?(\n|$)
```

---

Yeah they would be

---

Ah so it was the `> ![NOTE]` alert syntax specifically that was triggering not the content of the alert? I guess we could also include the other alert types here (`!TIP`, `!IMPORTANT`, `!WARNING`, `!CAUTION`). I don't entirely understand the regex here - it looks like Value uses [`regexp2`](https://github.com/dlclark/regexp2) which is based on [Go `regexp`](https://pkg.go.dev/regexp/syntax), but after reading the docs I'm still entirely sure what `(?s)` bit is doing here (appears to be setting a flag to let `.` match `\n` but not sure why this is needed and whether it applies just to the group?). I think something like

```
^ *> \[!(NOTE|IMPORTANT|WARNING|CAUTION|TIP)\] *$
```
would avoid unintentionally capturing other characters with `.` while still allowing for white space before or after alert syntax (could also use `\w` in place of ` ` but not sure we actually want to allow for other white space types).

---

Just spotted that you've already covered all altert types in last commit 73fd5c1 - I still don't entirely understand the regex but it works ~~so marking as resolved~~! (EDIT: I hadn't refreshed before commenting so hadn't seem Sam's comment either).

---

I looked at the `regex101` guide here: https://vale.sh/docs/keys/blockignores. I've tested both regexes and they both work. But if you understand yours, I'll go with that.

---

Also, I assume you meant `\s` in place of ` `? Hadn't occurred to me that `\s` was any type of whitespace. This is just in this flavour of regex? I've always taken it to mean strictly ` `.

---

Ah bad news, yours doesn't work

---

So in the [example linked to in the docs](https://regex101.com/r/mFM0kZ/1/) there it looks like it's excluding the whole block `{< file }` delimited block. Are we trying to similarly ignore the context of callouts or just the syntax which starts an alert? From https://regex101.com/r/rvRsMy/1 it looks like we just match the initial bit not the content. If so I think the regex I suggested will do so as well but is tighter / less chance for false matches

---

Went with `(?s)> \[!(CAUTION|IMPORTANT|NOTE|TIP|WARNING)\](\n|$)` and tested on `regex101`

---

Yeah we want to just ignore the top line itself, not the contents within it. Your regex doesn't work, feel free to try it yourself locally (run `vale sync; vale .`)